tkp/main.py

```diff
@@ -19,70 +20,82 @@
                             group_per_timestep
                             )
 from tkp.db.configstore import store_config, fetch_config
-from tkp.steps.persistence import create_dataset, store_images
+from tkp.steps.persistence import create_dataset, store_images_in_db
 import tkp.steps.forced_fitting as steps_ff
 from tkp.steps.varmetric import execute_store_varmetric
 
 
 logger = logging.getLogger(__name__)
 
 
-def run(job_name, supplied_mon_coords=[]):
+def setup(job_name, supplied_mon_coords=None):
+    """
+    Initialises the pipeline run.
+    """
+    if not supplied_mon_coords:
+        supplied_mon_coords = []
+
     pipe_config = initialize_pipeline_config(
         os.path.join(os.getcwd(), "pipeline.cfg"),
         job_name)
 
     # get parallelise props. Defaults to multiproc with autodetect num cores
-    parallelise = pipe_config.get('parallelise', {})
-    distributor = os.environ.get('TKP_PARALLELISE', parallelise.get('method',
-                                                                    'multiproc'))
-    runner = Runner(distributor=distributor,
-                    cores=parallelise.get('cores', 0))
-
+    distributor = os.environ.get('TKP_PARALLELISE', pipe_config.parallelise.method)
+    runner = Runner(distributor=distributor, cores=pipe_config.parallelise.cores)
 
-    #Setup logfile before we do anything else
+    # Setup logfile before we do anything else
     log_dir = pipe_config.logging.log_dir
-    setup_logging(log_dir,
-                  debug=pipe_config.logging.debug,
-                  use_colorlog=pipe_config.logging.colorlog
-                  )
+    setup_logging(log_dir, debug=pipe_config.logging.debug,
+                  use_colorlog=pipe_config.logging.colorlog)
 
     job_dir = pipe_config.DEFAULT.job_directory
     if not os.access(job_dir, os.X_OK):
         msg = "can't access job folder %s" % job_dir
         logger.error(msg)
-        raise IOError(msg)
+        sys.exit(1)
     logger.info("Job dir: %s", job_dir)
 
     db_config = get_database_config(pipe_config.database, apply=True)
     dump_database_backup(db_config, job_dir)
 
     job_config = load_job_config(pipe_config)
-    se_parset = job_config.source_extraction
-    deruiter_radius = job_config.association.deruiter_radius
-    beamwidths_limit = job_config.association.beamwidths_limit
-    new_src_sigma = job_config.transient_search.new_source_sigma_margin
+    dump_configs_to_logdir(log_dir, job_config, pipe_config)
+
+    sync_rejectreasons(tkp.db.Database().Session())
+
+    job_config, dataset_id = initialise_dataset(job_config, supplied_mon_coords)
+
+    return job_dir, job_config, pipe_config, runner, dataset_id
```

Comment:
That's a pretty long tuple to pass around - I know it's only used once, but for general sanity, maybe worth creating a throwaway namedtuple struct - e.g. `JobInit` (?) to put it in?